app/PaymentDrivers/Stripe/ACH.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com)
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2020. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://opensource.org/licenses/AAL
+ */
+
+namespace App\PaymentDrivers\Stripe;
+
+use App\Events\Payment\PaymentWasCreated;
+use App\Jobs\Mail\PaymentFailureMailer;
+use App\Jobs\Util\SystemLogger;
+use App\Models\ClientGatewayToken;
+use App\Models\GatewayType;
+use App\Models\Invoice;
+use App\Models\Payment;
+use App\Models\PaymentType;
+use App\Models\SystemLog;
+use App\PaymentDrivers\StripePaymentDriver;
+use Stripe\Exception\InvalidRequestException;
+
+class ACH
+{
+    public $stripe;
+
+    public function __construct(StripePaymentDriver $stripe)
+    {
+        $this->stripe = $stripe;
+    }
+
+    public function authorizeView(array $data)
+    {
+        return render('gateways.stripe.ach.authorize', array_merge($data));
+    }
+
+    public function authorizeResponse($request)
+    {
+        $state = [
+            'server_response' => json_decode($request->gateway_response),
+            'gateway_id' => $request->gateway_id,
+            'gateway_type_id' => $request->gateway_type_id,
+            'is_default' => $request->is_default,
+        ];
+
+        $customer = $this->stripe->findOrCreateCustomer();
+
+        $this->stripe->init();
+
+        $local_stripe = new \Stripe\StripeClient(
+            $this->stripe->company_gateway->getConfigField('apiKey')
+        );
+
+        try {
+            $local_stripe->customers->createSource(
+                $customer->id,
+                ['source' => $state['server_response']->token->id]
+            );
+        } catch (InvalidRequestException $e) {
+            return back()->with('ach_error', $e->getMessage());
+        }
+
+        $payment_meta = $state['server_response']->token->bank_account;
+        $payment_meta->brand = ctrans('texts.ach');
+        $payment_meta->type = ctrans('texts.bank_transfer');
+        $payment_meta->verified_at = null;
+        $payment_meta->btok = $state['server_response']->token->id;
```

Comment:
instead of btok, lets use a full readable variable bank_token or even just token